File: server/typescript/packages/metadata/src/presentation/view/view-schema.ts
// View attribute schemas — per-subtype attr inventories for view types.
// Consumed by registerCoreTypes().

import type { AttrSchema } from "../../registry.js";
import { ATTR_SUBTYPE_STRING } from "../../core/attr/attr-constants.js";
import {
  VIEW_CURRENCY_ATTR_LOCALE,
  VIEW_CURRENCY_ATTR_LOCALE_DEFAULT,
} from "./view-constants.js";

/** Attrs on view.currency. */
export const currencyViewAttrs: AttrSchema[] = [
  {
    name: VIEW_CURRENCY_ATTR_LOCALE,
    valueType: ATTR_SUBTYPE_STRING,
    required: false,
    default: VIEW_CURRENCY_ATTR_LOCALE_DEFAULT,
    description:
      "BCP 47 locale code controlling currency display formatting. Defaults to 'en-US' when omitted.",
  },
];

File: server/csharp/MetaObjects/Presentation/View/ViewSchema.cs
// View attribute schemas — attrs on view.currency.
//
// Colocated per ADR-0003. Mirrors typescript/packages/metadata/src/presentation/view/view-schema.ts.

using MetaObjects.Core.Attr;

namespace MetaObjects.Presentation.View;

/// <summary>Attribute schemas for the view concern.</summary>
public static class ViewSchema
{
    /// <summary>Attrs on view.currency.</summary>
    public static readonly IReadOnlyList<AttrSchema> CurrencyViewAttrs =
    [
        new AttrSchema(
            Name: ViewConstants.VIEW_CURRENCY_ATTR_LOCALE,
            ValueType: AttrConstants.ATTR_SUBTYPE_STRING,
            Required: false,
            Default: ViewConstants.VIEW_CURRENCY_ATTR_LOCALE_DEFAULT,
            Description: "BCP 47 locale code controlling currency display formatting. Defaults to 'en-US' when omitted."),
    ];
}
